terraform/modules/emr/templates/emr/check_status.sh

```diff
@@ -1,70 +1,40 @@
 #!/bin/bash
 DATA_PRODUCT=$1
-DEPENDANT_DATA_PRODUCT=$2
-DURATION=$3
+DATA_SOURCE=$2
 
-if [ -z "${DURATION}" ]
-  then 
-    END_TIME=80 #Seconds    ##Polls for 5 Hours then exits with Failure
-  else
-    END_TIME=${DURATION}    ##Polls for variable duration then exits with Failure
-fi
-
-INTERVAL=40 #Seconds     ##Checks for file flag file every 5 Minutes
-
-DATE=$(date '+%Y-%m-%d')
-
-while ((${SECONDS} < ${END_TIME}))
-do
-  #Get the status of the latest pdm run
-  DEPENDANT_DATA_PRODUCT_STATUS=$(hive -S -e "USE audit; SELECT Status FROM data_pipeline_metadata_hive dp INNER JOIN (
-   SELECT MAX(to_date(dateproductrun)) AS run_date FROM data_pipeline_metadata_hive WHERE dataproduct = '${DEPENDANT_DATA_PRODUCT}') md
-   ON to_date(dp.dateproductrun) = md.run_date
-    WHERE dp.dataproduct = '${DEPENDANT_DATA_PRODUCT}';")
-  if [[ ${DEPENDANT_DATA_PRODUCT_STATUS^^} == "COMPLETED" ]]
-  then
-   break
-  else
-    echo "$DEPENDANT_DATA_PRODUCT is running... Still waiting."
-  fi
-  sleep ${INTERVAL}
-done
-
-if [[ ${DEPENDANT_DATA_PRODUCT_STATUS^^} != "COMPLETED" ]]
-then
- echo "PDM did not finish. Exiting as a Failure - Please check previous dependencies for errors."
- exit 1
-fi
-
-#Get the correlation_id of the latest PDM run
-  LATEST_COR_ID=$(hive -S -e "USE audit; SELECT correlation_id FROM data_pipeline_metadata_hive dp INNER JOIN (
-   SELECT MAX(to_date(dateproductrun)) AS run_date FROM data_pipeline_metadata_hive WHERE dataproduct = '${DEPENDANT_DATA_PRODUCT}') md
-   ON to_date(dp.dateproductrun) = md.run_date
-    WHERE dp.dataproduct = '${DEPENDANT_DATA_PRODUCT}' AND upper(dp.status) = 'COMPLETED';") 
+  #Read correlation id
+  LATEST_COR_ID=`cat ~/${DATA_SOURCE}_CORRELATION_ID.txt`
 
   #RUN_ID will not be set if the data product did not run after PDM has completed
-  RUN_ID=$(hive -S -e "USE audit; SELECT run_id from data_pipeline_metadata_hive WHERE dataproduct = '${DATA_PRODUCT}' AND upper(status) = 'COMPLETED' AND correlation_id = '${LATEST_COR_ID}';")
+  RUN_ID=$(hive -e "USE audit; SELECT run_id from data_pipeline_metadata_hive WHERE dataproduct = '${DATA_PRODUCT}' AND upper(status) = 'COMPLETED' AND correlation_id = '${LATEST_COR_ID}';")
+  echo "RUN_ID is ${RUN_ID} "
 
-  #Check for FAILED run
-  RUN_STATUS=$(hive -S -e "USE audit; SELECT status FROM data_pipeline_metadata_hive WHERE dataproduct = '${DATA_PRODUCT}' AND upper(status) = 'FAILED' AND correlation_id = '${LATEST_COR_ID}';")
+  #Get run status
+  RUN_STATUS=$(hive -e "USE audit; SELECT status FROM data_pipeline_metadata_hive WHERE dataproduct = '${DATA_PRODUCT}' AND correlation_id = '${LATEST_COR_ID}';")
```

Comment:
Dont we need to specify the RUN_ID here? Otherwise won't the output of the SQL be a list of status for each run?